uniq/src/main.rs

```diff
@@ -0,0 +1,101 @@
+use std::{
+    fs::File,
+    io::{self, BufRead, BufReader, Read, Write},
+    process,
+};
+
+mod cli;
+
+fn main() {
+    let matches = cli::create_app().get_matches();
+
+    let input_filename = matches.value_of("INPUT").unwrap_or("-");
+    let output_filename = matches.value_of("OUTPUT").unwrap_or("-");
+
+    let reader: Box<dyn io::Read> = if input_filename == "-" {
+        Box::new(io::stdin())
+    } else {
+        Box::new(File::open(input_filename).unwrap_or_else(|err| {
+            eprintln!("uniq: Cannot open '{}' for reading: {}", input_filename, err);
+            process::exit(1);
+        }))
+    };
+    let mut reader = BufReader::new(reader);
+
+    let mut writer: Box<dyn io::Write> = if output_filename == "-" {
+        Box::new(io::stdout())
+    } else {
+        Box::new(File::create(output_filename).unwrap_or_else(|err| {
+            eprintln!("uniq: Cannot create '{}' for writing: {}", output_filename, err);
+            process::exit(1);
+        }))
+    };
+
+    if let Err(_) = uniq(&mut reader, &mut writer) {
+        process::exit(1);
+    };
+}
+
+type Result<T> = std::result::Result<T, io::Error>;
```

Comment:
You could just use `io::Result`, that are defined the same way as here, instead of creating a new type here